### PR TITLE
Update electrum

### PR DESCRIPTION
**Pull request by ghost**
todo: update binaries

### CODE REVIEW

**Comment by decryp2kanon** [comment]
@y-chan @volbil could you review this PR? i cannot due lack of experience of electrum... :sweat_smile:

**Review by akshaynexus — APPROVED** [review]
LGTM

File: electrum/constants.py [hunk 1]
```diff
@@ -78,7 +78,7 @@ class BitcoinMainnet(AbstractNet):
         'p2wsh':       0x02aa7ed3,  # Zpub
     }
     XPUB_HEADERS_INV = inv_dict(XPUB_HEADERS)
-    BIP44_COIN_TYPE = 34230
```
Comment (decryp2kanon) [inline electrum/constants.py:81]: AFAIK, 34230 is the mainnet port? @y-chan

Comment (y-chan) [reply electrum/constants.py:81]: This number is not mainport.
Used to be the same as the mainport.

**Review by y-chan — APPROVED** [review]
LGTM

File: electrum/constants.py (hunk 1 above)
Comment (decryp2kanon) [reply electrum/constants.py:81]: it was just a random number?

Comment (y-chan) [reply electrum/constants.py:81]: No.
This number is written this documents.
https://github.com/satoshilabs/slips/blob/master/slip-0044.md

**Comment by ghost** [comment]
Rebased with the updated checkpoints on master

File: electrum/constants.py (hunk 1 above)
Comment (cryptozeny) [reply electrum/constants.py:81]: i see. thanks to clarification!

**Comment by ghost** [comment]
Rebased with the fixed CoinGecko fiat conversion

**Comment by ghost** [comment]
Testnet support has been added

**Comment by ghost** [comment]
Fixed windows build from linux.

Currently android builds are broken and I don't know why.

**Comment by ghost** [comment]
Android build log

```
[INFO]:    Copying pyjnius java class to classes build dir
[INFO]:    -> directory context /home/user/wspace/electrum-sugar/.buildozer/android/platform/build/build/other_builds/pyjnius-python3-sdl2/armeabi-v7a__ndk_target_21/pyjnius
[INFO]:    -> running cp -a jnius/src/org /home/user/wspace/electrum-sugar/.buildozer/android/platform/build/build/javaclasses/Electrum
[INFO]:    <- directory context /opt/python-for-android
[INFO]:    Postbuilding sugar_yespower for armeabi-v7a
[INFO]:    Postbuilding android for armeabi-v7a
[INFO]:    Postbuilding kivy for armeabi-v7a
[INFO]:    # Installing pure Python modules
[INFO]:    *** PYTHON PACKAGE / PROJECT INSTALL STAGE ***
[INFO]:    The requirements (plyer) don't have recipes, attempting to install them with pip
[INFO]:    If this fails, it may mean that the module has compiled components and needs a recipe.
[INFO]:    -> directory context /home/user/wspace/electrum-sugar/.buildozer/android/platform/build/build
[INFO]:    -> running virtualenv --python=python3 venv
[INFO]:    Upgrade pip to latest version
[INFO]:    -> running bash -c source venv/bin/activate && pip install -U pip
Traceback (most recent call last):
  File "/usr/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/usr/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/opt/python-for-android/pythonforandroid/toolchain.py", line 1186, in <module>
    main()
  File "/opt/python-for-android/pythonforandroid/toolchain.py", line 1180, in main
    ToolchainCL()
  File "/opt/python-for-android/pythonforandroid/toolchain.py", line 663, in __init__
    getattr(self, args.subparser_name.replace('-', '_'))(args)
  File "/opt/python-for-android/pythonforandroid/toolchain.py", line 152, in wrapper_func
    build_dist_from_args(ctx, dist, args)
  File "/opt/python-for-android/pythonforandroid/toolchain.py", line 205, in build_dist_from_args
    args, "ignore_setup_py", False
  File "/opt/python-for-android/pythonforandroid/build.py", line 576, in build_recipes
    ignore_setup_py=ignore_project_setup_py
  File "/opt/python-for-android/pythonforandroid/build.py", line 644, in run_pymodules_install
    ), _env=copy.copy(base_env))
  File "/opt/python-for-android/pythonforandroid/logger.py", line 177, in shprint
    output = command(*args, **kwargs)
  File "/home/user/.local/lib/python3.6/site-packages/sh.py", line 1474, in __call__
    extracted_call_args, kwargs = self._extract_call_args(kwargs)
  File "/home/user/.local/lib/python3.6/site-packages/sh.py", line 1377, in _extract_call_args
    raise TypeError("Invalid special arguments:\n\n%s\n" % exc_msg)
TypeError: Invalid special arguments:

  'env': env must be a dict. Got environ({'USER': 'user', 'LANGUAGE': 'en_US.UTF-8', 'ANDROID_SDK_HOME': '/opt/android/android-sdk', 'HOSTNAME': '73f8586760fe', 'P4A_DEBUG_KEYSTORE': '/home/user/wspace/electrum-sugar/contrib/android_debug.keystore', 'WORK_DIR': '/home/user/wspace', 'SHLVL': '1', 'HOME': '/home/user', 'OLDPWD': '/home/user/wspace/electrum-sugar/electrum/gui/kivy', 'ANDROID_SDK_BUILD_TOOLS_VERSION': '28.0.3', 'P4A_DEBUG_KEYALIAS': 'electrum', 'ANDROIDMINAPI': '21', 'MAKEFLAGS': '', 'HOME_DIR': '/home/user', 'ANDROID_SDK_TOOLS_VERSION': '4333796', 'MAKE_TERMERR': '/dev/pts/0', 'ANDROID_SDK_TOOLS_ARCHIVE': 'sdk-tools-linux-4333796.zip', '_': '/usr/bin/make', 'ANDROIDAPI': '28', 'ANDROID_NDK_VERSION': '17c', 'TERM': 'xterm', 'P4A_DEBUG_KEYSTORE_PASSWD': 'unsafepassword', 'ANDROIDNDK': '/opt/android/android-ndk-r17c', 'PATH': '/opt/android/android-ndk-r17c/toolchains/arm-linux-androideabi-4.9/prebuilt/linux-x86/bin/:/opt/android/android-ndk-r17c/toolchains/arm-linux-androideabi-4.9/prebuilt/linux-x86_64/bin/:/opt/android/android-ndk-r17c:/opt/android/android-sdk/tools:/home/user/.buildozer/android/platform/apache-ant-1.9.4/bin:/home/user/.local/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin', 'ANDROID_NDK_ARCHIVE': 'android-ndk-r17c-linux-x86_64.zip', 'ANDROID_SDK_TOOLS_DL_URL': 'https://dl.google.com/android/repository/sdk-tools-linux-4333796.zip', 'MAKELEVEL': '1', 'USE_SDK_WRAPPER': '1', 'P4A_DEBUG_KEYALIAS_PASSWD': 'unsafepassword', 'ANDROID_NDK_HOME_V': '/opt/android/android-ndk-r17c', 'LANG': 'en_US.UTF-8', 'ANDROIDSDK': '/opt/android/android-sdk', 'ANDROID_NDK_DL_URL': 'https://dl.google.com/android/repository/android-ndk-r17c-linux-x86_64.zip', 'MAKE_TERMOUT': '/dev/pts/0', 'PACKAGES_PATH': '/home/user/.buildozer/android/packages', 'PWD': '/opt/python-for-android', 'LC_ALL': 'en_US.UTF-8', 'ANDROID_HOME': '/opt/android', 'MFLAGS': '', 'ANDROID_NDK_HOME': '/opt/android/android-ndk', 'GRADLE_OPTS': "-Xmx1536M -Dorg.gradle.jvmargs='-Xmx1536M'", 'VERSION_kivy': '82d561d62577757d478df52173610f925c05ecab', 'PYTHONPATH': '/home/user/wspace/electrum-sugar/.buildozer/android/platform/build/build/python-installs/Electrum'})

# Command failed: /usr/bin/python3 -m pythonforandroid.toolchain create --dist_name=Electrum --bootstrap=sdl2 --requirements=python3,android,openssl,plyer,kivy==82d561d62577757d478df52173610f925c05ecab,libffi,libsecp256k1,sugar_yespower,pyblake2 --arch armeabi-v7a --copy-libs --local-recipes /home/user/wspace/electrum-sugar/contrib/local_recipes --color=always --storage-dir="/home/user/wspace/electrum-sugar/.buildozer/android/platform/build" --ndk-api=21
# ENVIRONMENT:
#     USER = 'user'
#     LANGUAGE = 'en_US.UTF-8'
#     ANDROID_SDK_HOME = '/opt/android/android-sdk'
#     HOSTNAME = '73f8586760fe'
#     P4A_DEBUG_KEYSTORE = '/home/user/wspace/electrum-sugar/contrib/android_debug.keystore'
#     WORK_DIR = '/home/user/wspace'
#     SHLVL = '1'
#     HOME = '/home/user'
#     OLDPWD = '/home/user/wspace/electrum-sugar/electrum/gui/kivy'
#     ANDROID_SDK_BUILD_TOOLS_VERSION = '28.0.3'
#     P4A_DEBUG_KEYALIAS = 'electrum'
#     MAKEFLAGS = ''
#     HOME_DIR = '/home/user'
#     ANDROID_SDK_TOOLS_VERSION = '4333796'
#     MAKE_TERMERR = '/dev/pts/0'
#     ANDROID_SDK_TOOLS_ARCHIVE = 'sdk-tools-linux-4333796.zip'
#     _ = '/usr/bin/make'
#     ANDROID_NDK_VERSION = '17c'
#     TERM = 'xterm'
#     P4A_DEBUG_KEYSTORE_PASSWD = 'unsafepassword'
#     PATH = '/home/user/.buildozer/android/platform/apache-ant-1.9.4/bin:/home/user/.local/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin'
#     ANDROID_NDK_ARCHIVE = 'android-ndk-r17c-linux-x86_64.zip'
#     ANDROID_SDK_TOOLS_DL_URL = 'https://dl.google.com/android/repository/sdk-tools-linux-4333796.zip'
#     MAKELEVEL = '1'
#     USE_SDK_WRAPPER = '1'
#     P4A_DEBUG_KEYALIAS_PASSWD = 'unsafepassword'
#     ANDROID_NDK_HOME_V = '/opt/android/android-ndk-r17c'
#     LANG = 'en_US.UTF-8'
#     ANDROID_NDK_DL_URL = 'https://dl.google.com/android/repository/android-ndk-r17c-linux-x86_64.zip'
#     MAKE_TERMOUT = '/dev/pts/0'
#     PWD = '/home/user/wspace/electrum-sugar'
#     LC_ALL = 'en_US.UTF-8'
#     ANDROID_HOME = '/opt/android'
#     MFLAGS = ''
#     ANDROID_NDK_HOME = '/opt/android/android-ndk'
#     GRADLE_OPTS = "-Xmx1536M -Dorg.gradle.jvmargs='-Xmx1536M'"
#     PACKAGES_PATH = '/home/user/.buildozer/android/packages'
#     ANDROIDSDK = '/opt/android/android-sdk'
#     ANDROIDNDK = '/opt/android/android-ndk-r17c'
#     ANDROIDAPI = '28'
#     ANDROIDMINAPI = '21'
#
# Buildozer failed to execute the last command
# The error might be hidden in the log above this error
# Please read the full log, and search for it before
# raising an issue with buildozer itself.
# In case of a bug report, please add a full log with log_level = 2
Makefile:20: recipe for target 'apk' failed
make: [apk] Error 1 (ignored)
make[1]: Entering directory '/home/user/wspace/electrum-sugar/electrum/gui/kivy'
# Cleaning up
# rename main.py to electrum
# remove buildozer.spec
make[1]: Leaving directory '/home/user/wspace/electrum-sugar/electrum/gui/kivy'
~/wspace/electrum-sugar

```

**Comment by ghost** [comment]
Cherry-picked some updates from upstream to build android apk

**Comment by volbil** [comment]
I think it would be nice to have official explorer by default here https://github.com/sugarchain-project/electrum-sugar/pull/4/commits/588d1bfe61a5db828f8fd639d47260cb56a95943. Also https://sugar.wtf explorer can be added :)
Everything else looks good.

**Comment by cryptozeny** [comment]
i agreed to volbil.

**Comment by ghost** [comment]
@volbil @cryptozeny Addressed the review

**Comment by volbil** [comment]
Looking good @jangchongchan20

**Pull request closed without merging**